Always use no-quirks mode for CSS.supports()

annevk opened the pull request:
#### a21e36ac280d7e9918b0fa43094bbf37da3c01d9
<pre>
Always use no-quirks mode for CSS.supports()
<a href="https://bugs.webkit.org/show_bug.cgi?id=257643">https://bugs.webkit.org/show_bug.cgi?id=257643</a>
rdar://110162243

Reviewed by Tim Nguyen.

Align with the standard and other browsers. In particular

  <a href="https://drafts.csswg.org/css-color-4/#quirky-color">https://drafts.csswg.org/css-color-4/#quirky-color</a>
  <a href="https://drafts.csswg.org/css-values-4/#deprecated-quirky-length">https://drafts.csswg.org/css-values-4/#deprecated-quirky-length</a>

mandate this and those are the only quirks applying to CSS.supports().
As stated there this is not applicable to @supports.

* LayoutTests/css3/css-supports-document-parser-context-expected.txt:
* LayoutTests/css3/css-supports-document-parser-context.html:
* LayoutTests/css3/resources/css-supports-document-parser-context-nonstrict.html:
* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/web-platform-tests/quirks/classname-query-after-sibling-adoption.html:
* LayoutTests/imported/w3c/web-platform-tests/quirks/supports-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/quirks/text-decoration-doesnt-propagate-into-tables/standards-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/quirks/text-decoration-doesnt-propagate-into-tables/standards-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/quirks/text-decoration-doesnt-propagate-into-tables/standards.html:
* LayoutTests/tests-options.json:
* Source/WebCore/css/DOMCSSNamespace.cpp:
(WebCore::DOMCSSNamespace::supports):

Canonical link: <a href="https://commits.webkit.org/264835@main">https://commits.webkit.org/264835@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dcb628304419fcef65fea1ae44511e76ca122b63

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8785 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9073 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/9290 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10438 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8773 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/8793 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11060 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9043 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11632 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8931 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/9936 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/7797 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/10596 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7233 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8034 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/15536 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/8343 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8183 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11514 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/8665 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7067 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7926 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/7942 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2127 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12138 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/8412 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Looks good, but:
1. Do you need to do the same for `@supports` (which uses a different codepath)?
2. Can you add a link to the spec in the commit message?

annevk committed:
Always use no-quirks mode for CSS.supports()
https://bugs.webkit.org/show_bug.cgi?id=257643
rdar://110162243

Reviewed by Tim Nguyen.

Align with the standard and other browsers. In particular

  https://drafts.csswg.org/css-color-4/#quirky-color
  https://drafts.csswg.org/css-values-4/#deprecated-quirky-length

mandate this and those are the only quirks applying to CSS.supports().
As stated there this is not applicable to @supports.

* LayoutTests/css3/css-supports-document-parser-context-expected.txt:
* LayoutTests/css3/css-supports-document-parser-context.html:
* LayoutTests/css3/resources/css-supports-document-parser-context-nonstrict.html:
* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/web-platform-tests/quirks/classname-query-after-sibling-adoption.html:
* LayoutTests/imported/w3c/web-platform-tests/quirks/supports-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/quirks/text-decoration-doesnt-propagate-into-tables/standards-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/quirks/text-decoration-doesnt-propagate-into-tables/standards-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/quirks/text-decoration-doesnt-propagate-into-tables/standards.html:
* LayoutTests/tests-options.json:
* Source/WebCore/css/DOMCSSNamespace.cpp:
(WebCore::DOMCSSNamespace::supports):

Canonical link: https://commits.webkit.org/264835@main
